robot/client/client_test.go

```diff
@@ -98,6 +98,68 @@ var finalResources = []resource.Name{
 
 var pose1 = spatialmath.NewZeroPose()
 
+type mockRobotService struct {
+	pb.UnimplementedRobotServiceServer
+}
+
+func (ms *mockRobotService) ResourceNames(ctx context.Context, _ *pb.ResourceNamesRequest) (*pb.ResourceNamesResponse, error) {
+	board := board.Named("micro-rdk")
+	rNames := []*commonpb.ResourceName{
+		{
+			Namespace: string(board.API.Type.Namespace),
+			Type:      board.API.Type.Name,
+			Subtype:   board.API.SubtypeName,
+			Name:      board.Name,
+		},
+	}
+
+	return &pb.ResourceNamesResponse{Resources: rNames}, nil
+}
+
+func TestUnimplementedRPCSubtypes(t *testing.T) {
+	// create new robot service, the remote-side grpc server
+	logger := logging.NewTestLogger(t)
+	listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
Unfortunately we've had issues where parallel tests occupy the same port with code like this. Listening on `localhost:0` _should_ grab a random port, but sometimes two tests grab the same "random" port. Could you add some logic that looks like this and uses `TryReserveRandomPort`? I'll file a ticket to add some better test utilities for this, so we don't have to keep writing
```go
for i := 0; i < 10; i++ {
	port, err := utils.TryReserveRandomPort()
        ....
}
```

Reply:
Thanks for the tip! will use that here and in future tests